base/server.go

```diff
@@ -39,11 +39,19 @@ type Server struct {
 	multiDBDSN   string
 	multi        *multi
 
-	runOptions kbchat.RunOptions
+	runOptions    kbchat.RunOptions
+	serverOptions ServerOptions
 }
 
-func NewServer(name, announcement string, awsOpts *AWSOptions, multiDBDSN string, runOptions kbchat.RunOptions) *Server {
-	return &Server{
+type ServerOptions struct {
+	ReadMessagesFromSelf bool
+}
+
+func NewServer(
+	name, announcement string, awsOpts *AWSOptions, multiDBDSN string,
+	runOptions kbchat.RunOptions, serverOptions ...ServerOptions,
```

Comment:
I think this is cleaner https://github.com/keybase/client/blob/master/go/client/chat_api_ui.go#L23-L32